src/Flow/Mongo/MongoFile.php

```diff
@@ -90,12 +90,22 @@ public function saveChunk()
             ) {
                 throw new \Exception("Invalid upload! (size: {$actualChunkSize})");
             }
-            $chunk['data'] = new \MongoBinData($data);
+            $chunk['data'] = new \MongoBinData($data, 0); // \MongoBinData::GENERIC is not defined for older mongo drivers
             $this->config->getGridFs()->chunks->findAndModify($chunkQuery, $chunk, [], ['upsert' => true]);
             unlink($file['tmp_name']);
 
+            $this->ensureIndices();
+
             return true;
         } catch (\Exception $e) {
+            error_log("Could not store chunk: " . $e->getMessage() . "\n" . $e->getTraceAsString());
```

Comment:
hm - not sure I understand what you mean - in this method there is no access to any framework specific logging. This was meant to be similar to `\move_uploaded_file`, that issues a warning. Would it be better to user `trigger_error("Could not ...", E_USER_WARNING)`?